src/components/Dashboard/index.jsx

```diff
@@ -1,38 +1,79 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
 import './dashboard.scss';
 import Card from './Card';
+import { fetchRealTimeReport } from '../../redux/actions/realTimeReport';
+import listenToSocketEvent from '../../realTime';
+import ActivityFeed from './ActivityFeed';
 
 class Dashboard extends React.Component {
-  /*
-  Use case Example for dashboard card
-  component is a function that returns a component
-  Suppose you are working on Upselling Partners task
-  and you have created a class component named Upselling.
-  You will create a function renderUpselling that returns
-  your Upselling jsx component and pass it as props.
+  state = {
+    reportData: [],
+  };
+
+  componentDidMount = () => {
+    // triggers event to fetch data from the API
+    this.connectToSocket('newAutomation');
+  };
+
+  connectToSocket = (event) => {
+    listenToSocketEvent(event, (data) => {
+      this.handleDataFetch(data);
+    });
+  };
+
+  /**
+   * @description handles fetching of data and adds to the state
+   * @param {object} data coming in from the socket
+   * @returns {Void}
+   */
+  handleDataFetch = (data) => {
+    const { fetchUpdates } = this.props;
+    fetchUpdates(data);
```

Comment:
Please help me understand what `fetchUpdates` does?

Reply:
It triggers an action that takes the data received from the socket and adds it to the store

Reply:
Thanks for the clarity